src/ildasm/dis.cpp

```diff
@@ -1646,7 +1646,7 @@ BOOL Disassemble(IMDInternalImport *pImport, BYTE *ILHeader, void *GUICookie, md
 
             case ShortInlineBrTarget:
             {
-                char offset = (char) pCode[PC];
+                signed char offset = (signed char) pCode[PC];
```

Comment:
I searched some online, and it appears depending on machines, default char could be either signed or unsigned. I guess yours seems to be the latter.
Can you do one more testing by adding `-fsigned-char` under Unix path in CMakeLists.txt at the top of coreclr repo without any source change?
My concern is if singleness of char is inconsistent, we could have a similar issue in other place as well.

And also with regard to `long`, in fact we should use `int` which is 32 bit since MS compiler handles  `long` as `int` with the same size 32 bit while other platforms do not.